examples/cpp/topic_instances/PublisherApp.cpp

```diff
@@ -0,0 +1,392 @@
+    {
+        // Delete DDS entities contained within the DomainParticipant
+        participant_->delete_contained_entities();
+
+        // Delete DomainParticipant
+        DomainParticipantFactory::get_instance()->delete_participant(participant_);
+    }
+}
+
+void PublisherApp::on_publication_matched(
+        eprosima::fastdds::dds::DataWriter* /*writer*/,
+        const PublicationMatchedStatus& info)
+{
+    if (info.current_count_change == 1)
+    {
+        matched_ = info.current_count;
+        std::cout << "Publisher matched." << std::endl;
+        cv_.notify_one();
+    }
+    else if (info.current_count_change == -1)
+    {
+        matched_ = info.current_count;
+        std::cout << "Publisher unmatched." << std::endl;
+    }
+    else
+    {
+        std::cout << info.current_count_change
+                  << " is not a valid value for PublicationMatchedStatus current count change" << std::endl;
+    }
+}
+
+void PublisherApp::run()
+{
+    while (!is_stopped() && ((samples_ == 0) || !instances_sent_all_samples()))
+    {
+        publish();
+        // Wait for period or stop event
+        std::unique_lock<std::mutex> terminate_lock(mutex_);
+        cv_.wait_for(terminate_lock, std::chrono::milliseconds(period_ms_), [&]()
+                {
+                    return is_stopped();
+                });
+    }
+
+    // Wait for the timeout to expire if not stopped
+    if (!is_stopped() && timeout_s_ > 0)
+    {
+        std::cout << "Publisher stopping... waiting for " << timeout_s_ << "s to late-joiners" << std::endl;
+        std::unique_lock<std::mutex> timeout_lock(mutex_);
+        cv_.wait_for(timeout_lock, std::chrono::seconds(timeout_s_), [&]()
+                {
+                    return is_stopped();
+                });
+    }
+
+    // Dispose all instances if manually stopped
+    if (is_stopped())
+    {
+        for (int i = 0; i < instances_; i++)
+        {
+            writer_->dispose(&shapes_[instance_handles_[i]].first, instance_handles_[i]);
+            std::cout << shapes_[instance_handles_[i]].first.color() << " shape instance disposed" << std::endl;
+        }
+    }
+}
+
+void PublisherApp::publish()
+{
+    // Wait for the data endpoints discovery
+    std::unique_lock<std::mutex> matched_lock(mutex_);
+    cv_.wait(matched_lock, [&]()
+            {
+                // at least one has been discovered
+                return ((matched_ > 0) || is_stopped());
+            });
+    if (!is_stopped())
+    {
+        // Register instance check
+        bool instances_already_registered = (instance_handles_.size() > 0);
+
+        // Iterate per instance
+        for (int i = 0; i < instances_; i++)
+        {
+            InstanceHandle_t instance;
+            ShapeType shape_;
+            CLIParser::shape_configuration shape_configuration;
+            bool ret = false;
+
+            // Register instance if not already registered
+            if (!instances_already_registered)
```

Comment:
Why is the registration of the instances inside the public() method and not in the Constructor? I see that the operation is done only once at beginning, but can they change a run time?

Reply:
Good idea!